src/runtime/chan.go

```diff
@@ -37,11 +37,78 @@ func chanDebug(ch *channel) {
 	}
 }
 
+// blockchain is a chain of blocked operations
+type blockchain struct {
+	// next blocked operation
+	next *blockchain
+
+	// t is the task to wake up
+	// the pointer state holds the data buffer unless this is a select
+	// the data field should be set to zero if this is an rx op when the channel closes
+	t *task
+
+	// s is the channel select state corresponding to this channel operation on a select statement
+	// if nil, just use the data in the task
+	s *chanSelectState
+
+	// linked is the set of other blockchains that are tied to this blockchain
+	// when unblocking, these should all be detached appropriately
+	linked []blockchain
+}
+
+func (b *blockchain) ditch(old *blockchain) *blockchain {
```

Comment:
Missing comment.
Also, maybe it's just me but to me this function name sounds rather aggressive and doesn't clearly indicate what it does exactly.